test/lifecycle_test.dart

```diff
@@ -175,6 +175,29 @@ main() {
 
         sharedTypeTests(testSnapshotType);
       });
+
+      test('triggers error lifecycle events when an error is thrown', () {
+        var mountNode = new DivElement();
+        var renderedInstance = react_dom.render(components2.SetStateTest({}), mountNode);
+        LifecycleTestHelper component = getDartComponent(renderedInstance);
+        LifecycleTestHelper.staticLifecycleCalls.clear();
+        component.setState({"shouldThrow": true});
+
+        expect(component.lifecycleCalls, containsAllInOrder(['getDerivedStateFromError', 'componentDidCatch']));
+        expect(component.state["shouldThrow"], isFalse);
+      });
+
+      test('shows the correct error and info when an error is thrown', () {
```

Comment:
#nit could we change this test description to better reflect what we're actually testing, and not the implementation details of the test harness? Something like:
```suggestion
      test('passes the correct error/info to lifecycle methods when an error is thrown', () {
```